### PR TITLE
migrate 3.0.1 (new formula)

### PR DESCRIPTION
**Pull request by Mikulas**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR adds Formula for https://github.com/mattes/migrate

Closes https://github.com/mattes/migrate/issues/156

### CODE REVIEW

**Comment by Mikulas** [comment]
Hm, the test failed on fetch from a GitHub release, which is public. Maybe a rate limit issue?

File: Formula/migrate.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Migrate < Formula
+  desc "Database migrations. CLI and Golang library."
```
Comment (fxcoudert) [inline Formula/migrate.rb]: One sentence only, not final period

File: Formula/migrate.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  version "3.0.1"
+  sha256 "17e19678e930d4a48114937fb07300c107088d86a80c798630fabc08d32ee4ee"
+
+  def install
+    mv "migrate.darwin-amd64", "migrate"
+    bin.install "migrate"
+  end
+
+  test do
+    system "#{bin}/migrate", "-version"
```
Comment (fxcoudert) [inline Formula/migrate.rb]: We would also need a test that does more that output version number

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/migrate.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class Migrate < Formula
+  desc "Database migrations. CLI and Golang library."
+  homepage "https://github.com/mattes/migrate"
+  url "https://github.com/mattes/migrate/releases/download/v3.0.1/migrate.darwin-amd64.tar.gz"
+  version "3.0.1"
+  sha256 "17e19678e930d4a48114937fb07300c107088d86a80c798630fabc08d32ee4ee"
+
+  def install
+    mv "migrate.darwin-amd64", "migrate"
```
Comment (fxcoudert) [inline Formula/migrate.rb]: [Definitely not](https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Formulae.md#we-dont-like-binary-formulae) :)

> Our policy is that formulae in the core tap (homebrew/core) must be open-source and either built from source or produce cross-platform binaries (e.g. Java). Binary-only formulae should go to Homebrew Cask.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by Mikulas** [comment]
Thanks, I understand the comments. Unfortunately I didn't manage to get all the dependencies and build the binary locally. I will reopen this ticket if I find a way to fix it.

**Pull request closed without merging**